app/_hub/kong-inc/openid-connect/index.md

```diff
@@ -124,3145 +129,3071 @@ kong_version_compatibility:
-and Kong. That can also be used for authenticating the request. This is
-especially nice when used together with authorization code flow.
+<div class="endpoint delete indent">/openid-connect/jwks</div>
 
-*refresh_token*
+Deleting JWKS will also cause auto-generation of a new JWK set, thus it can be said that 
+the `DELETE` will actually cause a key rotation.
 
-This authentication method enables authentication using refresh or an offline
-token. It is similar to `password` and `client_credentials`, but uses different
-grant type when calling the token endpoint of the identity provider.
+##### Response
 
+```
+HTTP 204 No Content
+```
 
-#### config.client_id
+[jwk]: https://datatracker.ietf.org/doc/html/rfc7517#section-4
+[jwks]: https://datatracker.ietf.org/doc/html/rfc7517#appendix-A.1
+[json-web-key-set]: #json-web-key-set
 
-With this parameter, specify the `client_id` of your client
-that you have registered in OpenID Connect Provider. This
-is the client that authenticates Kong with your OpenID
-Provider. It can be considered a trusted client or confidential
-client. This parameter can also be an array as this plugin
-supports multiple clients. The right one is then picked up
-by matching to a request parameter (defined with `config.client_arg`).
 
-Default | Required
-:------:|:-------:
-`none`  | `no`
+---
 
-**Note:** In many cases, you will need to configure this to allow Kong
-to work as a trusted client. If you only do e.g. JWT bearer token
-verification, you won't need to configure this.
+## Preparations
 
+The OpenID Connect plugin relies in the most cases on a 3rd party identity provider.
+In this section we go through configuration of Keycloak and Kong.
 
-#### config.client_secret
+### Keycloak Configuration
 
-With this parameter, specify the `client_secret` of your client
-that you have registered in OpenID Connect Provider. As mentioned
-above, also this can be an array of client secrets.
+[Keycloak][keycloak] is used as the identity provider in the following examples.
+As the Keycloak is documented elsewhere, here is just a quick summary what
+we have done one that side (you can also use any other standard identity provider):
 
-Default | Required
-:------:|:-------:
-`none`  | `no`
+1. We created a confidential client `kong` with `private_key_jwt` authentication and pointed the
+   Keycloak to download the public keys from [the OpenID Connect Plugin JWKS endpoint][json-web-key-set]:
+   <br><br>
+   <img src="/assets/images/docs/openid-connect/keycloak-client-kong-settings.png">
+   <br>
+   <img src="/assets/images/docs/openid-connect/keycloak-client-kong-auth.png">
+   <br>
+2. We created another confidential client `service` with `client_secret_basic` authentication,
+   and the secret of `cf4c655a-0622-4ce6-a0de-d3353ef0b714` for which we enabled the client credentials grant:
+   <br><br>
+   <img src="/assets/images/docs/openid-connect/keycloak-client-service-settings.png">
```

Comment:
Missing redirect uri on this screenshot. Assuming http://service.test:8000/* like in the previous one.

Reply:
Service account does not need it.